packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -31,19 +34,28 @@ export const DesktopPlayer = ({
         // See https://gaurav5430.medium.com/css-flex-positioning-gotchas-child-expands-to-more-than-the-width-allowed-by-the-parent-799c37428dd6.
         style={{ display: 'flex', flexGrow: 1, overflow: 'hidden' }}
       />
-      <ProgressBarDesktop playerClient={playerClient} />
+      <ProgressBarDesktop playerClient={playerClient} durationMs={durationMs} />
     </StyledPlayer>
   );
 };
 
-export const ProgressBarDesktop = (props: { playerClient: PlayerClient }) => {
-  const { playerClient } = props;
+export const ProgressBarDesktop = (props: {
+  playerClient: PlayerClient;
+  durationMs: number;
+}) => {
+  const { playerClient, durationMs } = props;
+
+  const toHuman = (currentMs: number) => {
+    const date = new Date(null);
+    date.setMilliseconds(currentMs);
+    return date.toISOString().substr(14, 5);
+  };
```

Comment:
couple things:

1) my editor says `substr` is deprecated
2) `const date = new Date(currentMs)` is same as `date.setMilliseconds(currentMs)`
2) isn't `substr(14,5)` fragile? i looked up [toISOString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) and it says string can be 24 or 27 characters long. Instead i think this should be the equivalent, we have a new func called [dateToUtc](https://github.com/gravitational/webapps/blob/master/packages/shared/services/loc/loc.ts#L56), yassine added a new library called `date-fns` which you can pull out `format`:

```
format(dateInUTC, 'mm:ss') 
```

Reply:
> looked up [toISOString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) and it says string can be 24 or 27 characters long.

Thanks, I was just aping a stack overflow suggestion, but it looks like its broken. Fixed using your suggestion.